lib/nuget-parser/index.ts

```diff
@@ -78,26 +78,54 @@ function getFileContents(fileContentPath: string): string {
 // `dotnet` can publish the .deps file to a variety of places inside the publish folder, depending on what you're
 // including and targeting. Instead of trying different directories, just scan them all. In most cases, the file
 // will be in the root directory. (See https://github.com/Azure/azure-functions-vs-build-sdk/issues/518)
-function findDepsFilePathInPublishDir(dir: string): string | null {
+function findDepsFileInPublishDir(
+  dir: string,
+  filename: string,
+): null | Buffer {
+  // Try to get the file via full path.
+  const depsFilePath = path.join(dir, filename);
+  try {
+    const depsFile = fs.readFileSync(depsFilePath);
+    if (depsFile) return depsFile;
+  } catch (_) {
+    // Due to race conditions, fs docs suggests to not use .stat or .access to check if a file exists
+    // but instead we should to try and read it.
+    // https://nodejs.org/api/fs.html#fsstatpath-options-callback
+  }
+
+  // Otherwise, look in a nested dir for the same thing.
+  for (const subdir of fs.readdirSync(dir, { withFileTypes: true })) {
+    if (subdir.isDirectory()) {
+      const subdirPath = path.join(dir, subdir.name);
+      const result = findDepsFileInPublishDir(subdirPath, filename);
+      if (result) return result;
+    }
+  }
+
+  return null;
+}
+
+// Same as above but in edge cases, `dotnet` renames the file, ex: azure functions.
+function findDepsFileInPublishDirDefault(dir: string): Buffer | null {
```

Comment:
JS doesn't support overloading, so I came up with this
the solution it's a bit nasty, but we also need to default on cases where dotnet changes the name of the deps file, example: azure functions